cookbooks/inner_core_convection/inner_core_convection.cc

```diff
@@ -60,6 +79,32 @@ namespace aspect
 {
   namespace MaterialModel
   {
+    template <int dim>
+    double
+    InnerCore<dim>::
+    phase_function (const Point<dim> &position,
+                    const double temperature) const
+    {
+      // We need to convert the depth to pressure,
+      // keeping in mind that for this model, the pressure is only the dynamic pressure,
+      // so we have to compute the hydrostatic pressure explicitly as rho * g.
+      const double depth = this->get_geometry_model().depth(position);
+      const double reference_rho = 1.0; // formulation is nondimensionalized
+      const double pressure_gradient = this->get_gravity_model().gravity_vector(position).norm() * reference_rho;
+      double depth_deviation = depth - transition_depth;
+
+      if (std::abs(pressure_gradient) > 0)
+        depth_deviation -= transition_clapeyron_slope * (temperature - transition_temperature) / pressure_gradient;
+
+      double phase_func;
+      // use delta function for width = 0
+      if (transition_width == 0)
+        (depth_deviation > 0) ? phase_func = 1 : phase_func = 0;
```

Comment:
this is missing an assignment ``phase_func = ``

Reply:
Yes, I would agree -- legal, but hard to read :-)

Reply:
Oh, right. I just copied that from another material model. I'll make a separate pull request to fix it there too.